### PR TITLE
feat: nested ast processing

### PR DESCRIPTION
**Pull request by gratcliff**
## 🧰 What's being changed?
1. Adding support for deep-table AST processing. Functionally bringing values to the top child level within a `table` node.
2. Adding a new export, `astToPlainText` that returns plain text from an AST node.

## 🧪 Testing
This should not effect ReadMe's Hubs, no differences will appear there. Instead, you can test the lib's export methods via passing a fresh AST node and observing the result.

* [X ] **🆕 I'm adding something new!**
* [ ] **🐛 I'm fixing a bug!**
* [ ] **📸 I've made some changes to the UI!**

### CODE REVIEW

File: packages/markdown/components/Heading/index.jsx [hunk 1]
```diff
@@ -1,14 +1,5 @@
 const React = require('react');
 const PropTypes = require('prop-types');
-const kebab = require('lodash/kebabCase');
```
Comment (erunion) [inline packages/markdown/components/Heading/index.jsx:3]: Do we still need lodash?

Comment (gratcliff) [reply packages/markdown/components/Heading/index.jsx:3]: Yup, it's being used in section hash hrefs!

Comment (erunion) [reply packages/markdown/components/Heading/index.jsx:3]: Ah, missed that you moved it.

Comment (gratcliff) [reply packages/markdown/components/Heading/index.jsx:3]: In one of our plugin transformers.

File: packages/markdown/processor/compile/index.js [hunk 2]
```diff
@@ -4,3 +4,4 @@ export { default as rdmeEmbedCompiler } from './embed';
 export { default as rdmeVarCompiler } from './var';
 export { default as rdmeCalloutCompiler } from './callout';
 export { default as rdmePinCompiler } from './pin';
+export { default as plainTextCompiler } from './plain-text';
```
Comment (rafegoldberg) [inline packages/markdown/processor/compile/index.js:7]: I think of this folder as being exclusively for methods that modify the actual Unified `Compiler` instance, like so:

https://github.com/readmeio/api-explorer/blob/66eeca17c49cf30d27cbf1277ea7118bdac9a91a/packages/markdown/processor/compile/callout.js#L3

Can we move this `plain-text.js` script to the `plugin/` directory? Should work just as well from there!

File: packages/markdown/processor/plugin/section-anchor-id.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+const flatMap = require('unist-util-flatmap');
+const kebabCase = require('lodash/kebabCase');
+
+// Adds an empty <div id='section-slug'> next to all headings.
+// This is for compatibility with how we used to do slugs
+function transformer(ast) {
+  return flatMap(ast, node => {
+    if (node.type === 'element' && node.properties && node.properties.id) {
+      const id = `section-${kebabCase(node.properties.id)}`;
+      return [node, { type: 'element', tagName: 'div', properties: { id } }];
```
Comment (rafegoldberg) [inline packages/markdown/processor/plugin/section-anchor-id.js:10]: One of the issues with Dom's initial take on this is that when you used the backwards-compatible hash link, the page would scroll _past_ the actual header Ideally we'd move this in to the node's `children` array. Not sure, but something like this...

```js
  // etc.
  if (/* blah && blah */) {
    node.children.push({ type: 'element', tagName: 'div', properties: { id } });
  }
  return [node];
}
```

...might work?

File: packages/markdown/processor/plugin/table-flattening.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
+const flatMap = require('unist-util-flatmap');
+
+// Flattens table values and adds them as a seperate, easily-accessible key within children
```
Comment (rafegoldberg) [inline packages/markdown/processor/plugin/table-flattening.js:3]: ### 💖
be still, my heart

File: packages/markdown/processor/plugin/section-anchor-id.js (hunk 3 above)
Comment (gratcliff) [reply packages/markdown/processor/plugin/section-anchor-id.js:10]: Ahh that's interesting. Let me test it out!

File: packages/markdown/processor/compile/index.js (hunk 2 above)
Comment (gratcliff) [reply packages/markdown/processor/compile/index.js:7]: Happy to move it.

File: packages/markdown/processor/plugin/section-anchor-id.js (hunk 3 above)
Comment (gratcliff) [reply packages/markdown/processor/plugin/section-anchor-id.js]: Screen Shot 2020-05-05 at 5 44 30 PM

Comment (gratcliff) [reply packages/markdown/processor/plugin/section-anchor-id.js]: @rafegoldberg updated.

**Comment by gratcliff** [comment]
@erunion mind another once over and tagging a release if you're content?

**Comment by erunion** [comment]
```
Successfully published:
 - @readme/api-explorer@6.8.0
 - @readme/markdown@6.8.0
```

Make sure this doesn't go out before @rafegoldberg's 6.7.0 work in https://github.com/readmeio/readme/pull/2551

**Comment by gratcliff** [comment]
> ```
> Successfully published:
>  - @readme/api-explorer@6.8.0
>  - @readme/markdown@6.8.0
> ```
> 
> Make sure this doesn't go out before @rafegoldberg's 6.7.0 work in [readmeio/readme#2551](https://github.com/readmeio/readme/pull/2551)

Thanks! We're gonna jump a minor version in that PR.

**Pull request merged**